features/discovery/src/main/java/org/opennms/netmgt/discovery/messages/DiscoveryJob.java

```diff
@@ -98,4 +103,18 @@ public String toString() {
                  .add("ranges", m_ranges)
                  .toString();
     }
+    
+    /**
+     * <P>
+     * Returns int : the total task timeout for all ip ranges
+     * </P>
+     */
+    public int calculateTaskTimeout(){
+    	BigInteger taskTimeOut = BigInteger.ZERO;
+        for(final IPPollRange range : m_ranges) {        	
+        	taskTimeOut = taskTimeOut.add((BigInteger.valueOf(range.getRetries()).add(BigInteger.ONE)).multiply(range.getAddressRange().getSizeOfIpAddrRange()).multiply(BigInteger.valueOf(range.getTimeout())).multiply(FUDGE_FACTOR.toBigInteger()));
+        }
+        return taskTimeOut.intValue() > Integer.MAX_VALUE ? Integer.MAX_VALUE : taskTimeOut.intValue();
```

Comment:
This will not work because the Javadoc for BigInteger.intValue() says that it simply deletes all bits that are larger than the Integer limit. You need to compare taskTimeOut to BigInteger.valueOf(Integer.MAX_VALUE).